src/ui/errorBoundary/ErrorBoundary.tsx

```diff
@@ -0,0 +1,19 @@
+import { ErrorBoundary as ReactErrorBoundary } from 'react-error-boundary';
+import { ErrorInfo } from 'react';
+
+import { logger } from 'integrations/logger';
+
+import { ErrorBoundaryProps } from './ErrorBoundary.types';
+
+export const ErrorBoundary = ({ shouldLog = true, onError, ...props }: ErrorBoundaryProps) => {
+  const handleError = (error: Error, errorInfo: ErrorInfo) => {
+    onError?.(error, errorInfo);
+
+    if (shouldLog) {
+      logger.error(error);
+    }
```

Comment:
don't we need to place this line before `onError` call? In case `onError` interrupts further code execution? Just to be sure that logger has always logged the issue